src/main/java/com/example/bookstoreapp/services/impl/OrderServiceImpl.java

```diff
@@ -36,132 +30,184 @@
+
+        log.info("order amount: {} placed: {}, with item count: {} from cart: {}",
+                shoppingOrder.getTotalAmount(),
+                shoppingOrder.getId(),
+                shoppingOrder.getTotalItemCount(),
+                cart.getCartId());
+
+        cartItemService.clearCartPostOrder();
+        catalogItemService.reduceStockCount(itemPurchaseCount);
+
+        writeOrderAsJsonDocument(shoppingOrder);
+
+        return shoppingOrder;
+
+    }
+
+    @Override
+    public ShoppingOrder generateOrderInvoice(Long id) {
+        ShoppingOrderEntity savedOrder = shoppingOrderEntityRepository.findById(id).get();
+        if (savedOrder == null) throw new AppRuntimeException("No such order is present");
+
+        JSONObject orderJSONObject = new JSONObject(savedOrder);
+
+        String userHome = System.getProperty("user.home");
+        String fileName = savedOrder.getId() + ".pdf";
+        String filePath = userHome + "/" + fileName;
+
+        Document document = new Document();
+        try {
+            PdfWriter.getInstance(document, new FileOutputStream(filePath));
+        } catch (FileNotFoundException | DocumentException e) {
+            throw new RuntimeException(e);
+        }
+        document.open();
+
+        try {
+            Font fontBold = FontFactory.getFont(FontFactory.HELVETICA_BOLD, 12);
+
+            Paragraph header = new Paragraph("Order Invoice", fontBold);
+            header.setAlignment(Element.ALIGN_CENTER);
+            document.add(header);
+
+            document.add(new Paragraph("\n"));
+
+            SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+            document.add(new Paragraph("Order ID: " + orderJSONObject.getLong("id")));
+            document.add(new Paragraph("Order Date: " + dateFormat.format(savedOrder.getOrderDate())));
+            document.add(new Paragraph("Total Amount: " + orderJSONObject.getDouble("totalAmount")));
+            document.add(new Paragraph("Total Item Count: " + orderJSONObject.getInt("totalItemCount")));
+
+            document.add(new Paragraph("\n"));
+
+            document.add(new Paragraph("\n"));
+
+            JSONObject address = new JSONObject(orderJSONObject.getString("address"));
+            document.add(new Paragraph("Shipping Address: "));
+            document.add(new Paragraph(address.getString("line1")));
+            document.add(new Paragraph(address.getString("line2")));
+            document.add(new Paragraph(address.getString("city") + ", " + address.getString("state") + ", " + address.getString("country")));
+            document.add(new Paragraph("Postal Code: " + address.getString("postalCode")));
+
+        } catch (DocumentException e) {
+            throw new RuntimeException(e);
+        }
+        document.close();
+        return new ShoppingOrder().fromEntity(savedOrder);
+    }
+
+    private void writeOrderAsJsonDocument(ShoppingOrder shoppingOrder) {
```

Comment:
This method I have written , why it shows in your name

Reply:
Fixed